### PR TITLE
Fix assertions on parameterized test names for older Gradle versions

### PR DESCRIPTION
**Pull request by pshevche**
Fixing https://e.grdev.net/s/doszanvifrfno/tests/overview?outcome=FAILED

### CODE REVIEW

File: plugin/src/test/groovy/org/gradle/testretry/testframework/TestNGViaJUnitEngineFuncTest.groovy [hunk 1]
```diff
@@ -55,9 +58,12 @@ class TestNGViaJUnitEngineFuncTest extends BaseTestNGFuncTest {
 
     @Override
     String reportedParameterizedMethodName(String gradleVersion, String methodName, String paramType, int invocationNumber, @Nullable String paramValueRepresentation) {
-        GradleVersion.version(gradleVersion) > GRADLE_5_4_1
-            ? "${methodName}(${paramType}) > [${invocationNumber}] ${paramValueRepresentation ?: ''}"
-            : "${methodName}(${paramType})[${invocationNumber}]"
+        switch (GradleVersion.version(gradleVersion)) {
+            case { it < GRADLE_6_1 }: return "${methodName}(${paramType})[${invocationNumber}]"
+            case { it < GRADLE_7_0 }: return "[${invocationNumber}] ${paramValueRepresentation ?: ''}"
+            case { it < GRADLE_8_1 && it != GRADLE_7_6_4 }: return "${methodName}(${paramType})[${invocationNumber}]"
```
Comment (snoopcheri) [inline plugin/src/test/groovy/org/gradle/testretry/testframework/TestNGViaJUnitEngineFuncTest.groovy:64]: I assume that a potential Gradle 7.6.5 release would also match the pattern in line 64, but I have no idea how likely such a version would be. So it's probably not worth considering it at the moment.

**Review by snoopcheri — APPROVED** [review]

File: plugin/src/test/groovy/org/gradle/testretry/testframework/TestNGViaJUnitEngineFuncTest.groovy (hunk 1 above)
Comment (jean-andre-gauthier) [inline plugin/src/test/groovy/org/gradle/testretry/testframework/TestNGViaJUnitEngineFuncTest.groovy:64]: Wouldn't it be more resilient to use `it < GRADLE_8_1 && !(it >= GRADLE_7_6_4 && it < GRADLE_8_0)` as a condition?

**Review by jean-andre-gauthier — APPROVED** [review]

File: plugin/src/test/groovy/org/gradle/testretry/testframework/TestNGViaJUnitEngineFuncTest.groovy (hunk 1 above)
Comment (pshevche) [reply plugin/src/test/groovy/org/gradle/testretry/testframework/TestNGViaJUnitEngineFuncTest.groovy:64]: Yeah, I think we'll notice it during our CI checks and can easily fix it then.

Comment (pshevche) [reply plugin/src/test/groovy/org/gradle/testretry/testframework/TestNGViaJUnitEngineFuncTest.groovy:64]: I doubt there will be another 7.6.x release, so let's try to keep it simple for now.

**Pull request merged**